service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
+                    }
+
+                    auto tablet_id = new_data.new_tablet_map.first_tablet();
+                    while (true) {
+                        auto& tablet_info = new_data.new_tablet_map.get_tablet_info(tablet_id);
+                        auto last_token = new_data.new_tablet_map.get_last_token(tablet_id);
+                        updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
+                                                 .set_new_replicas(last_token, tablet_info.replicas)
+                                                 .set_stage(last_token, locator::tablet_transition_stage::allow_write_both_read_old)
+                                                 .set_transition(last_token, locator::tablet_transition_kind::rf_change)
+                                                 .build());
+
+                        if (auto next_tablet = new_data.new_tablet_map.next_tablet(tablet_id); next_tablet.has_value()) {
+                            tablet_id = *next_tablet;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+
+                updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
+                                                             .set_transition_state(topology::transition_state::tablet_migration)
+                                                             .set_version(_topo_sm._topology.version + 1)
+                                                             .del_global_topology_request()
+                                                             .del_global_topology_request_id()
+                                                             .build()));
+                updates.push_back(canonical_mutation(topology_request_tracking_mutation_builder(utils::UUID{req_uuid})
+                                                             .done(error)
+                                                             .build()));
+                unsigned topo_muts_count = updates.size();
+
+                if (new_data.status == tablet_replica_calculation_status::success) {
+                    std::map<sstring, sstring> opts(new_rf_per_dc.begin(), new_rf_per_dc.end());
+                    auto ks_md = keyspace_metadata::new_keyspace(ks_name, "org.apache.cassandra.locator.NetworkTopologyStrategy",
+                                                                 opts, ks.metadata()->initial_tablets());
+                    auto schema_muts = prepare_keyspace_update_announcement(_db, ks_md, guard.write_timestamp());
+                    for (auto &m: schema_muts)
+                        updates.emplace_back(m);
+                }
+
+                sstring reason = format("ALTER tablets KEYSPACE called with new RF-per-DC settings: {}", new_rf_per_dc);
+                rtlogger.info("do update {} reason {}", updates, reason);
```

Comment:
what?  and if `updates` contain 100K mutations?
we'll just print them at INFO level to the log?

Reply:
In other places we use log.trace, originally I had it like that, but temporarily changed to info and forgot to change it back - it's trace lvl now, thanks.